File: data-access/vueboard-repositories/Models/WorkItem.cs
namespace Vueboard.DataAccess.Models
{
  public class WorkItem : VueboardSoftDeleteEntity
  {
    public DateTime Created { get; set; } = DateTime.UtcNow;
    public DateTime Updated { get; set; } = DateTime.UtcNow;
    public int ProjectColumnId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Notes { get; set; }
    public string? FgColor { get; set; }
    public string? BgColor { get; set; }
    public int Index { get; set; }
    public ProjectColumn? ProjectColumn { get; set; }
    public List<WorkItemTag> WorkItemTags { get; set; } = new();
  }
}

File: web-client/app/types/work-item.ts
import * as zod from 'zod'

import { WorkItemOptionsSchema } from './work-item-options'

export const WorkItemSchema = WorkItemOptionsSchema.extend({
  uid: zod.uuidv4(),
  projectColumnId: zod.number()
    .int()
    .nonnegative()
    .optional(),
  id: zod.number()
    .int()
    .nonnegative()
    .optional(),
  index: zod.number()
    .int()
    .nonnegative(),
})

export type WorkItem = zod.infer<typeof WorkItemSchema>

export const DefaultWorkItemState: WorkItem = {
  uid: '',
  title: '',
  description: null,
  notes: null,
  workItemTags: [],
  fgColor: undefined,
  bgColor: undefined,
  index: 0
}
